cavern/src/intTest/java/org/opencadc/cavern/FilesPutTest.java

```diff
@@ -0,0 +1,318 @@
+            public Object run() throws Exception {
+                DataNode dn = new DataNode(nodeURI);
+                dn.getProperties().add(new NodeProperty(VOS.PROPERTY_URI_ISPUBLIC, Boolean.FALSE.toString()));
+                putNode(new URL(nodesURL.toExternalForm() + dn.getUri().getPath()), dn);
+                return null;
+            }
+        });
+
+        // Now attempt to PUT the file using /cavern/files - should still work
+        try {
+            Subject.doAs(cadcauthSubject, new PrivilegedAction<Object>() {
+                public Object run() {
+                    try {
+                        String filename = "src/test/resources/smallTextFile2.rtf";
+                        putFile(nodeURI, filename);
+                        return null;
+                    } catch (Throwable t) {
+                        throw new RuntimeException(t);
+                    }
+                }
+            });
+
+        } catch (Throwable t) {
+            log.error(t.getMessage(), t);
+            Assert.fail("Unexcepted exception: " + t.getMessage());
+        }
+    }
+
+
+    @Test
+    public void testPutFileNotAuth() throws Throwable {
+        final String uri = putTestURIStr + "/smallTextFile.rtf";
+        final VOSURI nodeURI = new VOSURI(uri);
+
+        try {
+            String filename = "src/test/resources/smallTextFile.rtf";
+            putFile(nodeURI, filename);
+            Assert.fail("should have had an access error.");
+
+        } catch (AccessControlException t) {
+            log.debug("expected & got AccessControlException");
+
+        } catch (Throwable t) {
+            log.error(t.getMessage(), t);
+            Assert.fail("Unexcepted exception: " + t.getMessage());
+        }
+    }
+
+
+    private void putFile(VOSURI uri, String fileName) throws Throwable {
+
+        RegistryClient regClient = new RegistryClient();
+        AuthMethod authMethod = AuthenticationUtil.getAuthMethod(AuthenticationUtil.getCurrentSubject());
+        if (authMethod == null) {
+            authMethod = AuthMethod.ANON;
+        }
+        URL filePutURL = regClient.getServiceURL(uri.getServiceURI(), Standards.VOSPACE_FILES_20, authMethod);
+        log.debug("baseURL for getFile: " + filePutURL.toExternalForm());
+        URL url = new URL(filePutURL.toString() + uri.getPath());
+        log.debug("requested url for putFile: " + url.toExternalForm());
+
+        File fileToUpload = new File(fileName);
+
+        ByteArrayOutputStream out = new ByteArrayOutputStream();
+        File tmp = new File(System.getProperty("java.io.tmpdir"));
+        HttpUpload put = new HttpUpload(fileToUpload, url);
+        put.setRequestProperty("Content-Type", "text/xml");
+
+        put.run();
+
+        if (put.getThrowable() != null) {
+            throw put.getThrowable();
+        }
+
+        Assert.assertEquals(200, put.getResponseCode());
+    }
+
+
+
+    private static void putNode(URL url, final Node node) throws Exception {
```

Comment:
You could have used the VOSpaceClient for this part, but that's okay, this works.

Reply:
ended up going this direction because md5sum was easier to test, and in all meant less code in FilesPutTest